packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/plugin-ledger-connector-quorum.ts

```diff
@@ -654,4 +710,80 @@ export class PluginLedgerConnectorQuorum
     }
     return this.runDeploy(req);
   }
+
+  // Low level function to call any method from web3.eth
+  // Should be used only if given functionality is not already covered by another endpoint.
+  public async invokeRawWeb3EthMethod(
+    args: InvokeRawWeb3EthMethodV1Request,
+  ): Promise<any> {
+    return new Promise((resolve, rejects) => {
+      this.log.debug("invokeRawWeb3EthMethod input:", JSON.stringify(args));
+
+      Checks.nonBlankString(
+        args.methodName,
+        "web3.eth method string must not be empty",
+      );
+
+      const looseWeb3Eth = this.web3.eth as any;
+      if (
+        Object.prototype.hasOwnProperty.call(looseWeb3Eth, args.methodName) &&
+        typeof looseWeb3Eth[args.methodName] === "function"
+      ) {
+        const web3Method = looseWeb3Eth[args.methodName];
+
+        const web3Response: Promise<unknown> = args.params
+          ? web3Method(...args.params)
+          : web3Method();
+
+        web3Response
+          .then((result: any) => {
+            this.log.debug("invokeRawWeb3EthMethod response (OK):", result);
+            return resolve(result);
+          })
+          .catch((err) => {
+            this.log.warn("invokeRawWeb3EthMethod response (ERROR):", err);
+            rejects(err);
+          });
+      } else {
+        rejects(new Error(`No method "${args.methodName}" in web3.eth`));
```

Comment:
@outSH Instead of returning two promises, could this be simplified to use the implicitly returned promise and then this `rejects` could be just a throw and there would be less identation/code in general?

Reply:
@petermetz See above